frontend/src/lib/WorkflowParser.ts

```diff
@@ -163,10 +163,17 @@ export default class WorkflowParser {
   // Returns a list of output paths for the entire workflow, by searching all nodes in
   // the workflow, and parsing outputs for each.
   public static loadAllOutputPaths(workflow: Workflow): StoragePath[] {
-    let outputPaths: StoragePath[] = [];
+    return this.loadAllOutputPathsWithStepNames(workflow).map(entry => entry[1]);
+  }
+
+  // Returns a list of object mapping a step name to output path for the entire workflow,
+  // by searching all nodes in the workflow, and parsing outputs for each.
+  public static loadAllOutputPathsWithStepNames(workflow: Workflow): Array<[string, StoragePath]> {
+    const outputPaths: Array<[string, StoragePath]> = [];
     if (workflow && workflow.status && workflow.status.nodes) {
       Object.keys(workflow.status.nodes).forEach(n =>
-        outputPaths = outputPaths.concat(this.loadNodeOutputPaths(workflow.status.nodes[n])));
+        this.loadNodeOutputPaths(workflow.status.nodes[n]).map(path =>
```

Comment:
I can't wait for a better way to iterate over values in a map like this :mask:

Reply:
I don't think we want an easier way. I found that lodash has `map` and `iteratee` methods that can be used here, but they don't actually look more readable.